src/db/transaction.rs

```diff
@@ -144,12 +147,25 @@ impl DbTransactionPool {
             }
         };
 
-        let (resp, db) = self.transaction_internal(check_precondition).await?;
+        let (mut resp, db) = self.transaction_internal(check_precondition).await?;
 
         // HttpResponse can contain an internal error
         match resp.error() {
             None => db.commit().await?,
-            Some(_) => db.rollback().await?,
+            Some(_) => {
```

Comment:
Unfortunately this `Some(_)` case doesn't cover many errors, only "internal" ones in HttpResponse (not even sure what creates those, maybe calls like `HttpResponse:: InternalServerError`? maybe nothing in this code base).

Most errors will short circuit through line 150 (via the `?`) -- I'm not sure how we can easily leave `Tags::extra` for those cases, I don't think we have a req/resp on hand 🤷‍♂️  -- I'd have to think about this more

Reply:
I'm kinda/sorta playing around with the idea of adding the tags to the error so that they get propagated up that way, but it's getting a bit messy right now. If I come up with something tomorrow, I'll add it.

Reply:
Added the tags to the passed-in request so they should automagically be added to the next sentry error call.